src/coreComponents/physicsSolvers/wavePropagation/AcousticWaveEquationSEM.cpp

```diff
@@ -312,20 +314,22 @@ void AcousticWaveEquationSEM::computeSeismoTrace( real64 const time_n, real64 co
     {
       if( receiverIsLocal[ircv] == 1 )
       {
-        p_rcvs[iSeismo][ircv] = 0.0;
-        real64 ptmpNp1 = 0.0;
-        real64 ptmpN = 0.0;
+        var_rcvs[iSeismo][ircv] = 0.0;
+        real64 vtmpNp1 = 0.0;
+        real64 vtmpN = 0.0;
         for( localIndex inode = 0; inode < receiverConstants.size( 1 ); ++inode )
         {
-          ptmpNp1 += pressure_np1[receiverNodeIds[ircv][inode]] * receiverConstants[ircv][inode];
-          ptmpN += pressure_n[receiverNodeIds[ircv][inode]] * receiverConstants[ircv][inode];
+          vtmpNp1 += var_at_np1[receiverNodeIds[ircv][inode]] * receiverConstants[ircv][inode];
+          vtmpN += var_at_n[receiverNodeIds[ircv][inode]] * receiverConstants[ircv][inode];
         }
-        //Temporary linear interpolation.
-        p_rcvs[iSeismo][ircv] = a1*ptmpN + a2*ptmpNp1;
+        // linear interpolation between the pressure value at time_n and time_(n+1)
```

Comment:
If I understand well you changed pressure by var as it can be another variable, maybe update the comment ?

Reply:
Exactly, the method is more general and could be later moved to WaveSolverBase, once the elastic solver is finished. I will update the comment.